### ISSUE TITLE
Adding wildcard ingress results in server config reloaded perpetually every 5 seconds

### ISSUE THREAD

**Issue by rosskevin** [containous/traefik#4187]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Added a wildcard ingress in `namespace: spinnaker` at `2018-11-12T16:48:19Z`.  Traefik is installed and running in `namespace: prod`.
 
`kubectl apply -f spinnaker-api.yml`
```yaml
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
    traefik.ingress.kubernetes.io/preserve-host: "true"
    traefik.ingress.kubernetes.io/priority: "100"
  labels:
    app: spin-gate
    ksonnet.io/component: spinnaker.api
  name: spin-gate
  namespace: spinnaker
spec:
  rules:
  - host: spinnaker-api.mydomain.com
    http:
      paths:
      - backend:
          serviceName: spin-gate
          servicePort: 8084
  tls:
  - hosts:
    - spinnaker-api.mydomain.com
    secretName: cert-wildcard-mydomain.com
```

### What did you expect to see?
A working ingress, calm logs, and a traefik that responds timely to service/ingress changes.

### What did you see instead?
Repeated logging every 5 seconds and eventually traefik stops responding to service/config changes (but continues serving).  In all cases, the traefik dashboard and whatever ingress I add works (even if I have to delete traefik to make it see the new service).

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.7.4
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

- GKE v1.11.2-gke.15
- traefik in the `prod` namespace
- cert-manager for acme wildcard certificates
- spinnaker-api ingress in the `spinnaker` namespace 

```toml

logLevel = "INFO"
sendAnonymousUsage = true
defaultEntryPoints = ["http","https"]
[entryPoints.http]
    address = ":8080"
    compress = true

[entryPoints.http.redirect]
    regex = "^http://(.*)"
    replacement = "https://$1"
    permanent = true

[entryPoints.https]
    address = ":8443"
    compress = true

[entryPoints.https.tls]
    minVersion = "VersionTLS11"
    cipherSuites = [
      "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
      "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
      "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
      "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
      "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
      "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
      "TLS_RSA_WITH_AES_256_GCM_SHA384",
      "TLS_RSA_WITH_AES_128_GCM_SHA256",
      "TLS_RSA_WITH_AES_128_CBC_SHA256",
      "TLS_RSA_WITH_AES_256_CBC_SHA",
      "TLS_RSA_WITH_AES_128_CBC_SHA",
    ]

[entryPoints.traefik]
    address = ":8888"          

[kubernetes]

[api]
    entryPoint = "traefik"
    dashboard = true    

[api.statistics]
    recentErrors = 10
```

```yaml
#  traefik.traefik
#  traefik.traefik-dashboard
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  labels:
    ksonnet.io/component: traefik.traefik
  name: traefik
rules:
  - apiGroups:
      - ''
    resources:
      - pods
      - services
      - endpoints
      - secrets
    verbs:
      - get
      - list
      - watch
  - apiGroups:
      - extensions
    resources:
      - ingresses
    verbs:
      - get
      - list
      - watch
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  labels:
    ksonnet.io/component: traefik.traefik
  name: traefik
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: traefik
subjects:
  - kind: ServiceAccount
    name: traefik
    namespace: prod
---
apiVersion: v1
data:
  traefik.toml: "logLevel = \"INFO\"\nsendAnonymousUsage = true\ndefaultEntryPoints
    = [\"http\",\"https\"]\n[entryPoints.http]\n    address = \":8080\"\n    compress
    = true\n\n[entryPoints.http.redirect]\n    regex = \"^http://(.*)\"\n    replacement
    = \"https://$1\"\n    permanent = true\n\n[entryPoints.https]\n    address = \":8443\"\n
    \   compress = true\n\n[entryPoints.https.tls]\n    minVersion = \"VersionTLS11\"\n
    \   cipherSuites = [\n      \"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\",\n      \"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256\",\n
    \     \"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305\",\n      \"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256\",\n
    \     \"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA\",\n      \"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA\",\n
    \     \"TLS_RSA_WITH_AES_256_GCM_SHA384\",\n      \"TLS_RSA_WITH_AES_128_GCM_SHA256\",\n
    \     \"TLS_RSA_WITH_AES_128_CBC_SHA256\",\n      \"TLS_RSA_WITH_AES_256_CBC_SHA\",\n
    \     \"TLS_RSA_WITH_AES_128_CBC_SHA\",\n    ]\n\n[entryPoints.traefik]\n    address
    = \":8888\"          \n\n[kubernetes]\n\n[api]\n    entryPoint = \"traefik\"\n
    \   dashboard = true    \n\n[api.statistics]\n    recentErrors = 10\n"
kind: ConfigMap
metadata:
  labels:
    ksonnet.io/component: traefik.traefik
  name: traefik-config
  namespace: prod
---
apiVersion: v1
kind: ServiceAccount
metadata:
  labels:
    ksonnet.io/component: traefik.traefik
  name: traefik
  namespace: prod
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: traefik
    ksonnet.io/component: traefik.traefik
  name: traefik
  namespace: prod
spec:
  externalTrafficPolicy: Cluster
  loadBalancerIP: 1.1.1.94
  ports:
    - name: http
      port: 80
      protocol: TCP
      targetPort: 8080
    - name: https
      port: 443
      protocol: TCP
      targetPort: 8443
  selector:
    app: traefik
  type: LoadBalancer
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: traefik
    ksonnet.io/component: traefik.traefik-dashboard
  name: traefik-dashboard
  namespace: prod
spec:
  ports:
    - port: 80
      targetPort: 8888
  selector:
    app: traefik
  type: ClusterIP
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: traefik
    ksonnet.io/component: traefik.traefik
  name: traefik
  namespace: prod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: traefik
  template:
    metadata:
      labels:
        app: traefik
    spec:
      containers:
        - image: traefik:1.7.4
          livenessProbe:
            failureThreshold: 3
            initialDelaySeconds: 10
            periodSeconds: 10
            successThreshold: 1
            tcpSocket:
              port: 8080
            timeoutSeconds: 2
          name: traefik
          ports:
            - containerPort: 8080
              name: http
            - containerPort: 8443
              name: https
            - containerPort: 8888
              name: admin
          readinessProbe:
            failureThreshold: 1
            initialDelaySeconds: 10
            periodSeconds: 10
            successThreshold: 1
            tcpSocket:
              port: 8080
            timeoutSeconds: 2
          resources:
            limits:
              cpu: 100m
              memory: 70Mi
            requests:
              cpu: 100m
              memory: 30Mi
          volumeMounts:
            - mountPath: /etc/traefik
              name: config-volume
      serviceAccountName: traefik
      terminationGracePeriodSeconds: 60
      volumes:
        - configMap:
            name: traefik-config
          name: config-volume
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/auth-secret: traefik-dashboard-admin-credentials
    ingress.kubernetes.io/auth-type: basic
    kubernetes.io/ingress.class: traefik
    traefik.ingress.kubernetes.io/preserve-host: 'true'
    traefik.ingress.kubernetes.io/priority: '100'
  labels:
    app: traefik-dashboard
    ksonnet.io/component: traefik.traefik-dashboard
  name: traefik-dashboard
  namespace: prod
spec:
  rules:
    - host: traefik.mydomain.com
      http:
        paths:
          - backend:
              serviceName: traefik-dashboard
              servicePort: 80
  tls:
    - hosts:
        - traefik.mydomain.com
      secretName: cert-wildcard-mydomain.com
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
time="2018-11-12T16:42:15Z" level=info msg="Using TOML configuration file /etc/traefik/traefik.toml"
time="2018-11-12T16:42:15Z" level=info msg="Traefik version v1.7.4 built on 2018-10-30_10:44:30AM"
time="2018-11-12T16:42:15Z" level=info msg="\nStats collection is enabled.\nMany thanks for contributing to Traefik's improvement by allowing us to receive anonymous information from your configuration.\nHelp us improve Traefik by leaving this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2018-11-12T16:42:18Z" level=info msg="Preparing server traefik &{Address::8888 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc0003cab00} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-11-12T16:42:18Z" level=info msg="Preparing server http &{Address::8080 TLS:<nil> Redirect:0xc000089b40 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:true ProxyProtocol:<nil> ForwardedHeaders:0xc0003caaa0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-11-12T16:42:18Z" level=info msg="Preparing server https &{Address::8443 TLS:0xc000291200 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:true ProxyProtocol:<nil> ForwardedHeaders:0xc0003caae0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-11-12T16:42:18Z" level=info msg="Starting server on :8888"
time="2018-11-12T16:42:18Z" level=info msg="Starting server on :8080"
time="2018-11-12T16:42:19Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-11-12T16:42:19Z" level=info msg="Starting server on :8443"
time="2018-11-12T16:42:19Z" level=info msg="Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null}"
time="2018-11-12T16:42:19Z" level=info msg="ingress label selector is: \"\""
time="2018-11-12T16:42:19Z" level=info msg="Creating in-cluster Provider client"
time="2018-11-12T16:42:19Z" level=error msg="Error configuring TLS for ingress prod/traefik-dashboard: secret prod/cert-wildcard-mydomain.com does not exist"
time="2018-11-12T16:42:20Z" level=error msg="Error configuring TLS for ingress prod/traefik-dashboard: secret prod/cert-wildcard-mydomain.com does not exist"
time="2018-11-12T16:42:20Z" level=error msg="Error configuring TLS for ingress prod/traefik-dashboard: secret prod/cert-wildcard-mydomain.com does not exist"
time="2018-11-12T16:42:20Z" level=error msg="Error configuring TLS for ingress prod/traefik-dashboard: secret prod/cert-wildcard-mydomain.com does not exist"
time="2018-11-12T16:42:21Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:42:21Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:42:21Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:42:25Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:42:25Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:42:25Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:42:30Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:42:30Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:42:30Z" level=info msg="Server configuration reloaded on :8443"


time="2018-11-12T16:48:19Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:19Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:19Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:19Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:19Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:21Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:21Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:21Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:21Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:21Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:30Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:30Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:30Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:30Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:30Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:33Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:33Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:33Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:33Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:33Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:34Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:34Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:34Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:34Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:34Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:45Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:45Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:45Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:45Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:45Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:46Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:46Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:46Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:46Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:46Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:48Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:48Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:48Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:48Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:48Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:48:51Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:51Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:48:51Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:48:51Z" level=info msg="Server configuration reloaded on :8443"
time="2018-11-12T16:48:51Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:49:05Z" level=warning msg="Into EntryPoint http, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:49:05Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.mydomain.com,mydomain.com). The new certificate will not be append to the EntryPoint."
time="2018-11-12T16:49:05Z" level=info msg="Server configuration reloaded on :8888"
time="2018-11-12T16:49:05Z" level=info msg="Server configuration reloaded on :8080"
time="2018-11-12T16:49:05Z" level=info msg="Server configuration reloaded on :8443"
```

**Comment by rosskevin** [comment]
I reproduced with debug logging and saved it as a gist so as not to clutter any conversation here:

https://gist.github.com/rosskevin/7d34a4b07c15f349478aeab0928b5b12

The process was:
- clean (no traefik, no spinnaker-api ingress)
- `kubectl apply -f traefik.yml`
- `kubectl apply -f spinnaker-api.yml`

**Comment by rosskevin** [comment]
I'm wondering if this is related to the difference in namespaces.  

1. traefik is in the`prod` namespace and seems fine as long as my ingresses are in that same namespace.  A `cert-wildcard-mydomain.com` exists in `prod` to serve these ingresses.
2. spinnaker is in the `spinnaker` namespace and we add the ingress there, expecting traefik in `prod` to be able to satisfy it which is successful, but logs get crazy.  Note tha a `cert-wildcard-mydomain.com` also exists in the `spinnaker` namespace for these ingresses.

While both namespaces have the same named secret `cert-wildcard-mydomain.com`, they are different as there is no cross access secrets of which I am aware.

Am I creating my ingress for spinnaker incorrectly?  Do I need to prefix it with the `prod` namespace?

**Comment by nmengin** [comment]
Hello @rosskevin ,

Many thanks for your interest in the project.

I can see that your secret which has to contain the certificate in the `prod` namespace seems to not exist:

`time="2018-11-12T17:12:47Z" level=error msg="Error configuring TLS for ingress prod/traefik-dashboard: secret prod/cert-wildcard-mydomain.com does not exist"`

Can you check if the `secret` exists in the namespace? 
If not, can you create it and give us a feddback?

Thanks in advance.

**Comment by rosskevin** [comment]
@nmengin this certificate does exist.  It pre-exists the traefik deployment and I have no idea why traefik continues to log that.  You can verify this based on the timestamp below vs the log above:

```
❯❯❯ kubectl describe certificate cert-wildcard-mydomain.com --namespace prod                           ✘ 1
Name:         cert-wildcard-mydomain.com
Namespace:    prod
Labels:       ksonnet.io/component=traefik.certWildcard
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"certmanager.k8s.io/v1alpha1","kind":"Certificate","metadata":{"annotations":{},"labels":{"ksonnet.io/component":"traefik.certWildcard"},...
API Version:  certmanager.k8s.io/v1alpha1
Kind:         Certificate
Metadata:
  Creation Timestamp:  2018-11-12T15:46:34Z
  Generation:          1
  Resource Version:    20797154
  Self Link:           /apis/certmanager.k8s.io/v1alpha1/namespaces/prod/certificates/cert-wildcard-mydomain.com
  UID:                 223e8d40-e692-11e8-8e60-42010a800198
Spec:
  Acme:
    Config:
      Dns 01:
        Provider:  cloudflare-mydomain.com
      Domains:
        *.mydomain.com
        mydomain.com
  Common Name:  *.mydomain.com
  Dns Names:
    mydomain.com
  Issuer Ref:
    Kind:       ClusterIssuer
    Name:       letsencrypt-prod
  Secret Name:  cert-wildcard-mydomain.com
Events:         <none>
```

**Comment by ReillyProcentive** [comment]
We're seeing this as well. Our use case is that we have a single Letsencrypt wildcard provisioned for our domain and then multiple ingresses (each with a separate subdomain) which reference said certificate. Upon startup Traefik emits the error about certificates not existing (even though they definitely do) once for each ingress (maybe a race condition?).

Trimmed logs:
```time="2018-12-11T20:04:04Z" level=info msg="Using TOML configuration file /config/traefik.toml"
time="2018-12-11T20:04:04Z" level=info msg="No tls.defaultCertificate given for https: using the first item in tls.certificates as a fallback."
time="2018-12-11T20:04:04Z" level=info msg="Traefik version 302849282c017b2fd63f70f997b4d61db1ad4e98 built on 2018-12-07_10:03:46PM"
time="2018-12-11T20:04:04Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2018-12-11T20:04:04Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:0xc0005d6580 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:0xc000539220 ForwardedHeaders:0xc0000c89a0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-12-11T20:04:04Z" level=info msg="Enabling ProxyProtocol for trusted IPs [100.64.0.0/10]"
time="2018-12-11T20:04:04Z" level=info msg="Preparing server https &{Address::443 TLS:0xc00028b0e0 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:0xc000539460 ForwardedHeaders:0xc0000c89c0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-12-11T20:04:04Z" level=info msg="Enabling ProxyProtocol for trusted IPs [100.64.0.0/10]"
time="2018-12-11T20:04:04Z" level=info msg="Preparing server admin &{Address::8080 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc0000c89e0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-12-11T20:04:04Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-12-11T20:04:04Z" level=info msg="Starting server on :443"
time="2018-12-11T20:04:04Z" level=info msg="Starting server on :80"
time="2018-12-11T20:04:04Z" level=info msg="Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"traffic = external\",\"IngressClass\":\"\",\"IngressEndpoint\":null}"
time="2018-12-11T20:04:04Z" level=info msg="Starting server on :8080"
time="2018-12-11T20:04:04Z" level=info msg="ingress label selector is: \"traffic=external\""
time="2018-12-11T20:04:04Z" level=info msg="Creating in-cluster Provider client"
time="2018-12-11T20:04:04Z" level=error msg="Error configuring TLS for ingress yawl/ingress-1: secret yawl/cert-ourdomain does not exist"
time="2018-12-11T20:04:04Z" level=error msg="Error configuring TLS for ingress yawl/ingress-2: secret yawl/cert-ourdomain does not exist"
time="2018-12-11T20:04:04Z" level=error msg="Error configuring TLS for ingress yawl/ingress-3: secret yawl/cert-ourdomain does not exist"
time="2018-12-11T20:04:04Z" level=error msg="Error configuring TLS for ingress yawl/ingress-4: secret yawl/cert-ourdomain does not exist"
...
...
repeated
...
...
time="2018-12-11T20:04:04Z" level=info msg="Server configuration reloaded on :8080"
time="2018-12-11T20:04:04Z" level=info msg="Server configuration reloaded on :80"
time="2018-12-11T20:04:04Z" level=info msg="Server configuration reloaded on :443"
... more "Error configuring TLS" logs ...
time="2018-12-11T20:04:06Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.our.domain). The new certificate will not be append to the EntryPoint."
time="2018-12-11T20:04:06Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.our.domain). The new certificate will not be append to the EntryPoint."
time="2018-12-11T20:04:06Z" level=warning msg="Into EntryPoint https, try to add certificate for domains which already have this certificate (*.our.domain). The new certificate will not be append to the EntryPoint."
... Logs repeat once per ingress. Then "Server configuration reloaded" messages. Then repeated again in 5 seconds.
```

Sample Ingress:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    external-dns.alpha.kubernetes.io/hostname: customer1.our.domain
    external-dns.alpha.kubernetes.io/target: ingress-e-o1.our.domain
    ingress.kubernetes.io/force-hsts: "true"
    ingress.kubernetes.io/hsts-include-subdomains: "true"
    ingress.kubernetes.io/hsts-max-age: "315360000"
    ingress.kubernetes.io/hsts-preload: "true"
    kubernetes.io/ingress.class: traefik
  labels:
    traffic: external
  name: ingress-customer1
spec:
  rules:
  - host: customer1.our.domain
    http:
      paths:
      - backend:
          serviceName: our-app-customer1
          servicePort: app
        path: /
  tls:
  - secretName: cert-ourdomain
```

Our `traefik.toml`
```
logLevel = "INFO"
checkNewVersion = false
defaultEntryPoints = ["https"]
[entryPoints]
  [entryPoints.http]
    address = ":80"
    [entryPoints.http.proxyProtocol]
      trustedIPs = ["100.64.0.0/10"]
    [entryPoints.http.redirect]
      entryPoint = "https"
  [entryPoints.https]
    address = ":443"
    [entryPoints.https.proxyProtocol]
      trustedIPs = ["100.64.0.0/10"]
    [entryPoints.https.tls]
      minVersion = "VersionTLS12"
      [[entryPoints.https.tls.certificates]]
        CertFile = "/ssl/wildcard.our1.domain/wildcard.our1.domain.crt"
        KeyFile = "/ssl/wildcard.our1.domain/wildcard.clinicaltm.com.key"
      [[entryPoints.https.tls.certificates]]
        CertFile = "/ssl/wildcard.our2.domain/wildcard.our2.domain.crt"
        KeyFile = "/ssl/wildcard.our2.domain/wildcard.our2.domain.key"
      [[entryPoints.https.tls.certificates]]
        CertFile = "/ssl/wildcard.our3.domain/wildcard.our3.domain.crt"
        KeyFile = "/ssl/wildcard.our3.domain/wildcard.our3.domain.key"
  [entryPoints.admin]
    address = ":8080"
[api]
  entrypoint = "admin"
[ping]
  entrypoint = "admin"
[metrics]
  [metrics.prometheus]
    entrypoint = "admin"
    buckets = [0.1,0.3,1.2,5.0]
[lifeCycle]
  requestAcceptGraceTimeout = "180s"
[kubernetes]
  labelselector = "traffic = external"
```

In our case we just switched from Comodo provided wildcard certs (mounted into the Traefik containers) to letsencrypt wildcard certs generated with cert-manager (and then referenced from Ingress objects). Interestingly as we've left the Comodo certs in for now the current behavior is that the Comodo certs are served instead of the letsencrypt ones (unless the Comodo certs are removed from the config) which is definitely not the expected behavior.

**Comment by rosskevin** [comment]
Single or multiple namespaces?

I feel like this should be a higher priority issue. This leaves one node at 30% perpetual usage for no reason.

**Comment by ReillyProcentive** [comment]
We have a single Traefik cluster in one namespace that hosts the ingress for multiple other namespaces. All ingresses that share a certificate are located in the same namespace though. We don't have any certificates that are duplicated across namespaces.

**Comment by rosskevin** [comment]
I posit that this is a multiple namespace issue.

**Comment by ReillyProcentive** [comment]
I've been taking a look at this for a while and I've found out a few things.

The issue specifically occurs if the exact same certificate is added to the certificate store for an entrypoint multiple times. It doesn't necessarily need to be a wildcard certificate and the certificate doesn't necessarily need to be loaded via multiple ingresses. I was also able to reproduce the issue as well by mounting the certificate into the Traefik container and then creating a single ingress that referenced an identical copy of the certificate.

I also took a look at what happens while the endless reload loop is happening. It appears that before each run a `"Received Kubernetes event kind *v1.Endpoints"` event is logged however there is no corresponding `"Skipping Kubernetes event kind *v1.Endpoints"` so it can be assumed that this is the cause of the configuration reload. I took a look at what my cluster was doing via `kubectl get endpoints --all-namespaces -w` and I saw typical leader election traffic typical of a k8s cluster. 

I then temporarily disabled some of the addon controllers we use which performed leader election/leasing and the issue continued to be present. At this point the only part of the k8s cluster that was emitting endpoint events was the leader election for the k8s API server itself. I then added `namespaces = ["namespace1", "namespace2"]` (the namespaces in our cluster that have Ingress objects) to the `[kubernetes]` section in order to force Traefik to watch just those namespaces for events instead of the equivalent of `--all-namespaces`. The reload loops stopped (!) after Traefik was restarted. I also took a look at adding `kube-system` to the list of watched namespaces and the issue started happening again.

The above seem to indicate that the reload loops are triggered by Endpoint events in the `kube-system` namespace, events which are ordinarily ignored as they do not affect the Traefik configuration. Taking a look at the code it seems likely to me that the duplicate cert issue is causing some kind of mismatch between the configuration that is currently applied vs the configuration that is calculated on every k8s event but it is strange that this only happens some of the time and not every time Traefik receives an event. 

I'm also not sure why Traefik is applying stale configurations either. While the issue is occurring we see that the configuration that is displayed in the "Configuration received from provider kubernetes" debug log message is correct however the configuration that is applied is stale and not the same as what is printed. This has caused service outages (luckily only on our dev cluster as we haven't yet rolled letsencrypt wildcard certs to prod yet) as Traefik continues to update the configuration with old backend endpoints even when those endpoints no longer exist (resulting in a "Bad Gateway" error for anyone trying to use the site). Perhaps this is caused by a buffer not being cleared out or an old pointer being passed around?

**Comment by ReillyProcentive** [comment]
@nmengin @juliens Can we have the priority on this bumped up given that the issue can cause service outage due to the stale configurations being applied?

@rosskevin Viable workarounds for this are to either restrict the Traefik controller to watching just the namespaces that you have ingresses in (hopefully you are following best practices and aren't deploying anything into `kube-system`) or to have your wildcard certificates generated in your Traefik namespace instead and then mounting them into the container (and specifying them from the config file). You would then omit the `tls` section entirely from your ingress definitions. TLS will still work as Traefik attaches the certificates to the entrypoint directly and just picks the correct one for a user based off of SNI. That approach has the downside that you will need to restart Traefik periodically to pick up renewed certificates as I do not believe Traefik will pick up changes to the cert automatically.

**Comment by zarqman** [comment]
I've been having the same reload problem. Per prior comments, I told Traefik to listen to a single namespace and confirmed that it's doing so via the debug logs. Further, Traefik itself is inside that same namespace as are all the certs. I'm still seeing the ongoing reloads.

I have observed that the order of both the "Server configuration reloaded on :<port>" and "Into EntryPoint https, try to add certificate for domains which already have this certificate..." log messages are in varying orders per execution. Is it possible the detection for a changed Kubernetes config is order dependent?

Traefik 1.7.5 + Kubernetes 1.11.5

**Comment by zarqman** [comment]
I have confirmed that this is not a namespace issue (although reducing triggered events can reduce the unnecessary reloads that happen when the config has not changed).

The underlying issue is inconsistent ordering of data returned by the Kubernetes API which in turn leads Traefik to compare differently-ordered configs and decide that they are different.

I've submitted PR #4307 which sorts the TLS configs which eliminates the excess config reloads. It also de-dups the TLS certs (for when multiple Ingresses reference the same TLS Secret--common when using wildcards), which resolves the unwarranted 'error adding cert' warnings.
